kqueen/engines/openstack_kubespray.py

```diff
@@ -756,6 +754,10 @@ def deprovision(self, volume_names):
                         self.c.remove_router_interface(router, port_id=i.id)
                         time.sleep(2)
                     self.c.delete_router(router.id)
+                    logger.debug("Router {0} has been deleted".format(self.stack_name))
+                if "resources" not in self.cluster.metadata:
+                    # Deployment failed before instance spawning
```

Comment:
logging? exception?

Reply:
exception was raised while provisioning, in that case we are handling deprovision of failed deployment

Reply:
ok, agree